lib/entities/workflow-definition.ts

```diff
@@ -14,14 +14,37 @@ import { wrapCollection } from '../common-utils'
 import enhanceWithMethods from '../enhance-with-methods'
 import { ActionType } from './role'
 
-/* Workflow Step */
+/* Workflow Step Action */
+export enum WorkflowStepActionType {
+  App = 'app',
+  Email = 'email',
+  Task = 'task',
+}
+export interface WorkflowStepAction<T = Record<string, any>> {
+  appId?: string
+  appActionId?: string
+  type: WorkflowStepActionType
+  configuration?: T
+}
 
-export type WorkflowStepAction = {
-  action: Link<'AppAction'>
+type EmailActionRecipient = string | Link<'User'> | Link<'Team'>
+
+export type WorkflowStepEmailActionConfiguration = {
+  recipients: EmailActionRecipient[]
+}
+
+export type WorkflowStepTaskActionConfiguration = {
+  assignee: Link<'User'> | Link<'Team'>
   body: string
-  headers: Record<string, unknown>
+  dueDate?: number
+}
+
+export type WorkflowStepAppActionConfiguration = {
+  body: Record<string, any>
```

Comment:
Both body and headers are optional params.

Reply:
done